File: 2023/tests/day22.test.js
const fs = require('fs');
const { part1, part2 } = require('../src/day22');

const example1 = `
1,0,1~1,2,1
0,0,2~2,0,2
0,2,3~2,2,3
0,0,4~0,2,4
2,0,5~2,2,5
0,1,6~2,1,6
1,1,8~1,1,9
`.trim();

const input = fs.readFileSync('inputs/input22.txt', 'utf8');

describe('part 1', () => {
  test('example', () => {
    expect(part1(example1)).toBe(5);
  });

  test('solution', () => {
    expect(part1(input)).toBe(441);
  });
});

describe('part 2', () => {
  test('example', () => {
    expect(part2(example1)).toBe(7);
  });

  test('solution', () => {
    expect(part2(input)).toBe(80778);
  });
});

File: 2024/AdventOfCode/Tests/Day22Test.cs
using NUnit.Framework;

namespace AdventOfCode.Tests;

[TestFixture]
public class Day22Test {
    private const string Example = "Inputs/22-Example.txt";
    private const string Example2 = "Inputs/22-Example2.txt";
    private const string Input = "Inputs/22.txt";
    private static Day22 _day;

    [TestFixture]
    public class Part1 {
        [Test]
        public void TestExample() {
            _day = new Day22(Example);
            const string expected = "37327623";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day22(Input);
            const string expected = "13004408787";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }

    [TestFixture]
    public class Part2 {
        [Test]
        public void TestExample2() {
            _day = new Day22(Example2);
            const string expected = "23";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day22(Input);
            const string expected = "1455";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }
}